src/puya/context.py

```diff
@@ -33,7 +33,9 @@ def try_get_source(self, location: SourceLocation | None) -> SourceMeta:
             start_line = end_line = location.line
             if location.end_line:
                 end_line = location.end_line
-            src_content = source_lines[start_line - 1 : end_line]  # location.lines is one-indexed
+            src_content = list(
+                source_lines[start_line - 1 : end_line]  # location.lines is one-indexed
```

Comment:
Would it make sense to just convert the line index to 0-based when we create the `SourceLocation` rather than having to remember this 'gotcha' everywhere we use it?

Reply:
I considered that, but the only time it matters is when matching up a `SourceLocation` to source code, which is done only here. Having the location info be different from the Python AST could also be a different kind of gotcha. The balance may swing towards zero indexed when we start working on additional languages, if the majority of those return zero indexed lines.